executor/adapter.go

```diff
@@ -167,7 +168,7 @@ func (a *ExecStmt) IsReadOnly() bool {
 // This function builds an Executor from a plan. If the Executor doesn't return result,
 // like the INSERT, UPDATE statements, it executes in this function, if the Executor returns
 // result, execution is done after this function returns, in the returned ast.RecordSet Next method.
-func (a *ExecStmt) Exec(ctx context.Context) (ast.RecordSet, error) {
+func (a *ExecStmt) Exec(goCtx goctx.Context, ctx context.Context) (ast.RecordSet, error) {
```

Comment:
The second `ctx` can be removed.
We can set `ctx` of `ExecStmt` at creation time.